### PR TITLE
Test cmd/server

### PR DESCRIPTION
**Pull request by lkysow**
a couple more tests to go

### CODE REVIEW

File: cmd/server.go [hunk 1]
```diff
@@ -100,77 +101,119 @@ type boolFlag struct {
...
-		viper.BindPFlag(f.name, serverCmd.Flags().Lookup(f.name))
+		c.Flags().Bool(f.name, f.value, f.description)
+		s.Viper.BindPFlag(f.name, c.Flags().Lookup(f.name))
+	}
+
+	return c
+}
+
+func (s *ServerCmd) preRun() error {
+	// if passed a config file then try and load it
```
Comment (anubhavmishra) [inline cmd/server.go]: As you pointed out that golang comments should be capitalized.  May be we can capitalize this.

File: cmd/server.go [hunk 2]
```diff
@@ -100,77 +101,119 @@ type boolFlag struct {
...
+	if err := setAtlantisURL(&config); err != nil {
+		return err
 	}
+	if err := setDataDir(&config); err != nil {
+		return err
+	}
+	sanitizeGithubUser(&config)
 
-	RootCmd.AddCommand(serverCmd)
+	// config looks good, start the server
```
Comment (anubhavmishra) [inline cmd/server.go]: same here.

File: cmd/server_test.go [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+		cmd.GHTokenFlag: "token",
+	})
+	err := c.Execute()
+	Ok(t, err)
+
+	Equals(t, "user", passedConfig.GithubUser)
+}
+
+func TestExecute_Flags(t *testing.T) {
+	t.Log("should use all flags that are set")
```
Comment (anubhavmishra) [inline cmd/server_test.go]: Let's be consistent with the use of capitalization on `t.log` statements. So here `should` -> `Should`

**Comment by lkysow** [comment]
@anubhavmishra good call on the standardization, fixed everything I found up (in these classes, we still aren't consistent with the t.Log statements anywhere)

**Comment by anubhavmishra** [comment]
LGTM! 👍

**Pull request merged**